hpx/runtime/get_thread_name.hpp

```diff
@@ -23,10 +23,6 @@ namespace hpx
     /// identifies the thread in the context of HPX. If the function is called
     /// while no HPX runtime system is active, the result will be "<unknown>".
     HPX_API_EXPORT std::string get_thread_name();
-
-    /// \cond NOINTERNAL
-    HPX_API_EXPORT hpx::util::itt::domain const& get_thread_itt_domain();
-    /// \endcond
 }
```

Comment:
Removing `HPX_EXPORT` from this function will cause linker errors on Windows and Mac/OS if the user references `util::annotated_function` in his/her code.

Reply:
This function was completely removed. But see other comments.

Reply:
Right, I didn't see that at first. I'm however concerned by making the `thread_specific_ptr` directly 'visible' through `HPX_EXPORT`. See comments below.